### PR TITLE
Show suggested registry keys instead of dumping all keys.

### LINKED ISSUE #249: plone.api.portal.get_registry_record dumps all registry entries.
plone.api.portal.get_registry_record dumps all registry entries when a non-existing key is passed in.
In Plone 5 this is a list of 1500+ items, including all plone.resources settings etc.

The list shown should be reduced to something like
```
records = [_name for _name, record in registry.records.items() if name.lower() in _name.lower()]
```

### PR DESCRIPTION
**Pull request by jaroel**
Closes #249

### CODE REVIEW

File: src/plone/api/tests/test_portal.py [hunk 1]
```diff
@@ -492,29 +492,40 @@ def test_get_invalid_registry_record(self):
...
 
-        # A selection of records which should exist in all plone versions
-        should_be_theres = (
-            "plone.app.discussion.interfaces.IDiscussionSettings.captcha",
-            "plone.app.querystring.field.Creator.title",
-            "plone.app.querystring.operation.int.largerThan.description",
-            "plone.app.querystring.operation.selection.is.widget",
-        )
+        # We don't have suggestions for this name.
+        self.assertTrue('Available records might be are:' not in exc_str)
```
Comment (mattss) [inline src/plone/api/tests/test_portal.py]: I think this should be: "Available records are".

Comment (mattss) [inline src/plone/api/tests/test_portal.py]: In fact, I think it should be 'Did you mean?' in line with the other tests below.

Comment (jaroel) [inline src/plone/api/tests/test_portal.py]: I check this in the next test, so I removed this one.

File: src/plone/api/portal.py [hunk 2]
```diff
@@ -255,14 +255,22 @@ def get_registry_record(name=None):
...
-        raise InvalidParameterError(
-            "Cannot find a record with name '{0}'.\n"
-            "Available records are:\n"
-            "{1}".format(name, '\n'.join(records))
-        )
+        # Show all records that 'look like' name.
+        # We don't dump the whole list, because it 1500+ items.
+        lower_name = name.lower()
+        records = [_name.lower() for _name, _ in registry.records.items()]
+        records = [_name for _name in records if lower_name in _name]
```
Comment (adamcheasley) [inline src/plone/api/portal.py]: Why not do this filtering in one list comprehension?

Comment (jaroel) [inline src/plone/api/portal.py]: pep8 80 chars max.
I've made it case-sensitive, so now it fits on one line :)

File: src/plone/api/tests/test_portal.py [hunk 3]
```diff
@@ -492,29 +492,37 @@ def test_get_invalid_registry_record(self):
...
+        self.assertFalse('Did you mean?:' in exc_str)
+
+        # Check with suggestions
+        with self.assertRaises(InvalidParameterError) as cm:
+            portal.get_registry_record(name='querystring')
+        exc_str = str(cm.exception)
+
+        # Check for an error with suggestions.
+        self.assertTrue(exc_str.startswith("Cannot find a record with name"))
+        self.assertTrue('Did you mean?:' in exc_str)
```
Comment (hvelarde) [inline src/plone/api/tests/test_portal.py]:
```python
self.assertIn('Did you mean?:', exc_str)
```

File: src/plone/api/tests/test_portal.py [hunk 4]
```diff
@@ -492,29 +492,37 @@ def test_get_invalid_registry_record(self):
...
-        for should_be_there in should_be_theres:
-            self.assertIn((should_be_there + '\n'), str(cm.exception))
+        # Check without suggestion
+        with self.assertRaises(InvalidParameterError) as cm:
+            portal.get_registry_record(name='a random unique string')
+        exc_str = str(cm.exception)
+
+        # Check for an error, but no suggestions.
+        self.assertTrue(exc_str.startswith("Cannot find a record with name"))
+        self.assertFalse('Did you mean?:' in exc_str)
```
Comment (hvelarde) [inline src/plone/api/tests/test_portal.py]: @jaroel is better:

```python
self.assertNotIn('Did you mean?:', exc_str)
```

Comment (jaroel) [inline src/plone/api/tests/test_portal.py]: What the hell is `self.assertNotIn`?
I don't understand or like self.assert[*].
Really, I'd rather just use
```python
assert 'Did you mean?:' not in exc_str
```

Comment (keul) [inline src/plone/api/tests/test_portal.py]: Not sure but I think that `assert` is deprecated

Comment (hvelarde) [inline src/plone/api/tests/test_portal.py]: @jaroel [assertNotIn](https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertNotIn) is a more readable (AKA pythonic) way of testing what you're testing

**Pull request merged**